src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -172,5 +163,27 @@ private static JToken ParseToJToken(string value)
 
             return state;
         }
+
+        private async Task<DurableOrchestrationStatus> GetDurableOrchestrationStatusAsync(OrchestrationState orchestrationState, bool showHistory, bool showHistoryInputOutput)
+        {
+            var history = string.Empty;
+            if (showHistory)
+            {
+                // TBD - parsing and processing of content
+                history = await this.client.GetOrchestrationHistoryAsync(orchestrationState.OrchestrationInstance);
+            }
+
+            return new DurableOrchestrationStatus
+            {
+                Name = orchestrationState.Name,
+                InstanceId = orchestrationState.OrchestrationInstance.InstanceId,
+                CreatedTime = orchestrationState.CreatedTime,
+                LastUpdatedTime = orchestrationState.LastUpdatedTime,
+                RuntimeStatus = (OrchestrationRuntimeStatus)orchestrationState.OrchestrationStatus,
+                Input = ParseToJToken(orchestrationState.Input),
+                Output = ParseToJToken(orchestrationState.Output),
+                History = ParseToJToken(history),
```

Comment:
It seems inefficient to fetch the history as a string and then format it into JSON here. Can we instead fetch it as JSON and assign it more directly?

Reply:
@cgillum I think this is limitation of the underlying Durable Task Framework and especially this method:

```csharp
public async Task<string> GetOrchestrationHistoryAsync(OrchestrationInstance instance)
        {
            if (instance == null || string.IsNullOrEmpty(instance.InstanceId) ||
                string.IsNullOrEmpty(instance.ExecutionId))
            {
                throw new ArgumentNullException("instance");
            }

            ThrowIfInstanceStoreNotConfigured();

            IEnumerable<OrchestrationHistoryEventEntity> eventEntities =
                await
                    tableClient.ReadOrchestrationHistoryEventsAsync(instance.InstanceId, instance.ExecutionId)
                        .ConfigureAwait(false);
            var events = new List<HistoryEvent>(eventEntities
                .OrderBy(ee => ee.SequenceNumber)
                .Select(historyEventEntity => historyEventEntity.HistoryEvent));

            string history = JsonConvert.SerializeObject(
                events, Formatting.Indented, new JsonSerializerSettings
                {
                    TypeNameHandling = TypeNameHandling.Objects
                });
            return history;
        }
```
It is a bit unfortunate that the data is retrieved  first as JSON but then the method decides to serialize it. 

Another way is to go directly to the storage table and retrieve the content directly but I prefer not to do it since Azure Storage Table in this case is an implementation detail and and in my opinion it is better to go via the Durable Task Framework for accessing the history. In the same time I am not able to find another method that gives access to the history content. Let me know if you have better suggestion.

Thank you!

Reply:
I agree we shouldn't go directly to table storage. What about simply doing [JArray.FromObject(events)](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JArray_FromObject.htm)?

Reply:
@cgillum Great! I will change and we will be able to follow the original data structure.